src/shims/foreign_items/posix.rs

```diff
@@ -66,7 +66,18 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
                 let buf = this.read_scalar(buf)?.not_undef()?;
                 let count = this.read_scalar(count)?.to_machine_usize(this)?;
                 let result = if fd == 0 {
-                    throw_unsup_format!("reading from stdin is not implemented")
+                    use std::io::{self, Read};
```

Comment:
Since reading from stdin can break non-determinism, please add a call to `this.check_no_isolation`.